state/cockroachdb/cockroachdb_access.go

```diff
@@ -36,6 +37,7 @@ const (
 	connectionStringKey        = "connectionString"
 	errMissingConnectionString = "missing connection string"
 	tableName                  = "state"
+	maxRetries                 = 5 // A bad driver connection error can occur inside the sql code so this essentially allows for more retries since the sql code does not allow that to be changed
```

Comment:
Rename this to `defaultMaxRetries` and please expose this as a new metadata option. Magic numbers are not good.

This can be done by updating the following struct to

```go
type cockroachDBMetadata struct {
	ConnectionString string
	TableName        string
	MaxRetries *int
}
```

Wherever you want to use `maxRetries`, you should check whether `p.metadata.MaxRetries` is an integer. If yes, use this value. If no, use the `defautlMaxRetries` value from your constant.

Reply:
Should this retry option also be applied to other functions, other than Ping?

Reply:
if it is not applied to other functions, can we please rename the metadata option since it will be misleading for the user to think they have configured retires for operations.